### ISSUE TITLE
Extensions Order Dependence in conf.py

### ISSUE THREAD

**Issue by Shaun-Griffith-Hive** [sphinx-doc/sphinx#4221]
Subject: The order of extensions listed in the `conf.py extensions` array can cause an error.

### Problem
- Napoleon extension listed before Autodoc extension causes an error, as autodoc is not handled first.

#### Procedure to reproduce the problem
```
sphinx-quickstart --quiet \
 --project='some project name' \
 --author=some_author  \
 -v 0.1  \
 --release 0.1.1  \
 --dot=_ \
 --ext-autodoc \
 --ext-todo \
 --ext-viewcode \
 --extensions=sphinx.ext.napoleon \
 --makefile \
 --no-batchfile
```

Workaround: specify the extensions in the `--extensions` list, with autodoc before napoleon. The extensions are emitted into `conf.py` in the order given in the option:
```
sphinx-quickstart --quiet \
 --project='some project name' \
 --author=some_author  \
 -v 0.1  \
 --release 0.1.1  \
 --dot=_ \
 --extensions=sphinx.ext.autodoc,sphinx.ext.todo,sphinx.ext.viewcode,sphinx.ext.napoleon \
 --makefile \
 --no-batchfile
```

#### Error logs / results
```
Extension error:
Unknown event name: autodoc-process-docstring
```

#### Expected results
```
Finished: An initial directory structure has been created.

You should now populate your master file ./index.rst and create other documentation
source files. Use the Makefile to build the docs, like so:
   make builder
where "builder" is one of the supported builders, e.g. html, latex or linkcheck.
```

### Environment info
- OS: Linux 16.04
- Python version: 3.6.3
- Sphinx version: 1.6.5

**Comment by anntzer** [comment]
I very recently reported a similar issue with numpydoc (https://github.com/numpy/numpydoc/issues/141).  I would assume that for this case, sphinx.ext.napoleon should just force-load sphinx.ext.autodoc at the very beginning.

**Comment by tk0miya** [comment]
Fixed at #4233 . Thank you for reporting!
